test/watershed.jl

```diff
@@ -0,0 +1,34 @@
+using ImageFiltering
+
+@testset "watershed" begin
+    
+    img = zeros(100, 100)
+    img[25:75, 25:75] = 1
+
+    img, _ = magnitude_phase(img)
+
+    markers = zeros(Int, size(img))
+    markers[1, 1] = 1
+    markers[50, 50] = 2
+
+    result = watershed(img, markers)
+
+    @test length(result.segment_labels) == 2
+    @test result.segment_labels == collect(1:2)
+    @test all(label->(label==result.image_indexmap[50,50]), result.image_indexmap[26:74,26:74])
+
+
+    img = ones(15, 15)
+    #minima of height 0.2
```

Comment:
It should be depth instead of height.
In the context of this algorithm, the depth of a minima is defined as the minimum vertical height one must go up starting from that minima to reach another minima.